examples/leetcode_agent.py

```diff
@@ -0,0 +1,67 @@
+import asyncio
+import os
+
+from dotenv import load_dotenv
+
+from notte.agents.falco.agent import FalcoAgent as Agent
+from notte.agents.falco.agent import FalcoAgentConfig as AgentConfig
+from notte.common.credential_vault.hashicorp.vault import HashiCorpVault
+from notte_integrations.remote_sessions.steel_pool import SteelBrowserPool
+
+# Load environment variables
+_ = load_dotenv()
+
+vault_url = os.getenv("VAULT_URL")
```

Comment:
I think there some code duplication with `vault_agent.py` now. Can you fix that ? Also I am thinking, could we automatically load `VAULT_URL and `VAULT_DEV_ROOT_TOKEN_ID` in `HashiCorpVault` it would make this simpler for users to use the vault